src/OpenTelemetry/Metrics/MetricReader.cs

```diff
@@ -63,6 +63,8 @@ public AggregationTemporality Temporality
         /// <summary>
         /// Attempts to collect the metrics, blocks the current thread until
         /// metrics collection completed, shutdown signaled or timed out.
+        /// Asynchronous instrument callbacks, if any, will be triggered
```

Comment:
Maybe copy the wording from the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#collect):

_If there are asynchronous instruments involved, their callback functions will be triggered._